File: 2022-2023/sourcecode/lab/lab12/InjectionBasicsJava/app/src/main/java/pl/edu/uwr/pum/injectionbasicsjava/computer/components/Monitor.java
package pl.edu.uwr.pum.injectionbasicsjava.computer.components;

import javax.inject.Inject;

import pl.edu.uwr.pum.injectionbasicsjava.computer.Computer;

public class Monitor {
    @Inject
    public Monitor() {
    }

    public String setComputer(Computer computer){
        return "monitor connected";
    }
}

File: 2022-2023/sourcecode/lab/lab12/InjectionBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/injectionbasicskotlin/computer/components/Monitor.kt
package pl.edu.uwr.pum.injectionbasicskotlin.computer.components

import pl.edu.uwr.pum.injectionbasicskotlin.computer.Computer
import javax.inject.Inject

class Monitor @Inject constructor() {
    fun setComputer(computer: Computer): String{
        return "monitor connected"
    }
}